src/pages/givingform.tsx

```diff
@@ -0,0 +1,834 @@
+}
+
+const emailAddressPattern = new RegExp(
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+)
+
+const datePattern = new RegExp(/\d\d\d\d-\d\d-\d\d/)
+
+function encodeStringField(fieldName: string, fieldValue?: string | undefined) {
+    if (fieldValue !== undefined) {
+        return `entry.${fieldNameToEntryId[fieldName]}=${encodeURI(fieldValue)}`
+    } else {
+        return undefined
+    }
+}
+
+function convertPayloadToGoogleFormUrl(
+    giftFormData: GiftFormData,
+    formId: string
+): string {
+    const baseUrl = "https://docs.google.com/forms/d/e/"
+
+    const formParts = []
+    formParts.push(encodeStringField("title", giftFormData.title))
+    formParts.push(encodeStringField("givenName", giftFormData.givenName))
+    formParts.push(encodeStringField("familyName", giftFormData.familyName))
+    formParts.push(
+        encodeStringField("streetAddress", giftFormData.streetAddress)
+    )
+    formParts.push(
+        encodeStringField("extraAddressLine", giftFormData.extraAddressLine)
+    )
+    formParts.push(encodeStringField("townCity", giftFormData.townCity))
+    formParts.push(encodeStringField("postCode", giftFormData.postCode))
+    formParts.push(
+        encodeStringField("telephoneNumber", giftFormData.telephoneNumber)
+    )
+    formParts.push(encodeStringField("emailAddress", giftFormData.emailAddress))
+    formParts.push(
+        encodeStringField(
+            "giftAmountInGBP",
+            giftFormData.giftAmountInGBP.toString()
+        )
+    )
+    formParts.push(encodeStringField("giftType", giftFormData.giftType))
+
+    formParts.push(
+        encodeStringField(
+            "regularGiftFrequency",
+            giftFormData.regularGiftFrequency !== undefined &&
+                giftFormData.regularGiftFrequency === "other"
+                ? giftFormData.otherRegularGiftRequency
+                : giftFormData.regularGiftFrequency
+        )
+    )
+
+    formParts.push(
+        encodeStringField(
+            "regularGiftCommencementDate",
+            giftFormData.regularGiftCommencementDate
+        )
+    )
+
+    formParts.push(
+        encodeStringField(
+            "thisGiftIsEligibleForGiftAid",
+            giftFormData.thisGiftIsEligibleForGiftAid ? "true" : "false"
+        )
+    )
+
+    formParts.push(
+        encodeStringField(
+            "allFutureGiftsAreEligibleForGiftAid",
+            giftFormData.allFutureGiftsAreEligibleForGiftAid ? "true" : "false"
+        )
+    )
+
+    formParts.push(
+        encodeStringField(
+            "retrospectivelyReclaimGiftAidForFourYears",
+            giftFormData.retrospectivelyReclaimGiftAidForFourYears
+                ? "true"
+                : "false"
+        )
+    )
+
+    const url =
+        baseUrl +
+        formId +
+        "/formResponse?" +
+        formParts.join("&") +
+        "&submit=SUBMIT"
```

Comment:
Not a big deal if this is already working, but for encoding stuff like this I'd normally use <https://github.com/sindresorhus/query-string>.

Reply:
Happy to use something like that. I've just been throwing this together.